File: URI/python/1040.py
N1, N2, N3, N4 = [float(x) for x in input().split()]

media = (2 * N1 + 3 * N2 + 4 * N3 + N4) / 10
print('Media: {:.1f}'.format(media))

if media < 5:
    print('Aluno reprovado.')
elif media > 7:
    print('Aluno aprovado.')
else:
    print('Aluno em exame.')
    exame = float(input())
    print('Nota do exame: {:.1f}'.format(exame))
    media_final = (exame + media) / 2
    if media_final >= 5:
        print('Aluno aprovado.')
    else: 
        print('Aluno reprovado.')
    print('Media final: {:.1f}'.format(media_final))

File: uri/c/1040.c
#include <stdio.h>

int main() {

    float N1, N2, N3, N4, media, exame;
    scanf("%f %f %f %f", &N1, &N2, &N3, &N4);

    media = (2 * N1 + 3 * N2 + 4 * N3 + 1 * N4) / 10;
    printf("Media: %.1f\n", media);

    if (media >= 7) {
        printf("Aluno aprovado.\n");
    } else if (media >= 5 && media < 7) {
        printf("Aluno em exame.\n");
        scanf("%f", &exame);
        printf("Nota do exame: %.1f\n", exame);
        media = (media + exame) / 2.0;

        if (media >= 5) {
            printf("Aluno aprovado.\n");
            printf("Media final: %.1f\n", media);
        } else {
            printf("Aluno reprovado.\n");
            printf("Media final: %.1f\n", media);
        }

    } else {
        printf("Aluno reprovado.\n");
    }

    return 0;

}